spec/requests/list_controller_spec.rb

```diff
@@ -11,11 +11,6 @@
   end
 
   describe '#index' do
-    it "doesn't throw an error with a negative page" do
-      get "/#{Discourse.anonymous_filters[1]}", params: { page: -1024 }
-      expect(response.status).to eq(200)
-    end
```

Comment:
@tgxworld Was there a reason for allowing negative page numbers?

Reply:
I think it was just to prevent errors being thrown in our logs. I think we can change it so that negative numbers will just be converted to 0?

Reply:
I disagree, our API should be returning HTTP 400 this is an invalid request not a 200, and definitely should not be returning a 500 which signifies to "crappy security scanners" that they are up to something so keep trying till you get a SQL injection jackpot